### PR TITLE
use an event handler rather than the passive command handler to log messages

### PR DESCRIPTION
**Pull request by ryanlerch**
<strike>Previously, we were doing some fudgery to log messages as the result of commands from the meetings plugin. This however, was not caused by the bot not being able to passively record messages it sends, but by the fact that command only, by default works on MessageType.TEXT, and when you send a message from a maubot plugin with evt.reply or evt.respond, it is hard coded to send the message as MessageType.NOTICE.</strike>

<strike>So this commit removes all the fudgery, and tells the passive logging command to also look at the NOTICE message type.</strike>

### CODE REVIEW

**Comment by ryanlerch** [comment]
Ok -- i'll need to rethink this -- its now logging messages from other bots (zodbot in my case) -- but now not logging its own messages -- may need to return the fudgery

**Comment by ryanlerch** [comment]
> Ok -- i'll need to rethink this -- its now logging messages from other bots (zodbot in my case) -- but now not logging its own messages -- may need to return the fudgery

figured it out! updated the commit (and the message):

```
use an event handler to log messages rather than passive handler
Previously, we were doing some fudgery to log messages as the result of
commands from the meetings plugin. (or from other maubot plugins)
This was caused by the passive handler not processing messages from the
bot itself, and the passive handler not processing messages of type
NOTICE (which maubot's evt.reply sends messages as)

So this commit removes all the fudgery, and adds a event handler on all
room messages of type NOTICE and TEXT to log messages.
```

File: meetings/__init__.py [hunk 1]
```diff
@@ -19,17 +19,24 @@
...
-    
+
 class Config(BaseProxyConfig):
   def do_update(self, helper: ConfigUpdateHelper) -> None:
     helper.copy("backend")
     helper.copy("backend_data")
     helper.copy("powerlevel")
+    helper.copy("tag_commands")
+    helper.copy("tag_commands_prefix")
+    helper.copy("tag_commands_start_only")
```
Comment (GregSutcliffe) [inline meetings/__init__.py]: These don't seem to match the keys in the base-config file (tags/tags_commandprefix/tags_commandatstart). I prefer these names though, should we update the base-config to match?

File: base-config.yaml [hunk 2]
```diff
@@ -3,3 +3,13 @@ powerlevel: 50
...
+tags:
+  action: "🚩"
+  info: "✏️"
+  agreed: "👍"
+  rejected: "👎"
+  idea: "💭"
+  halp: "🛟"
+  link: "🔗"
+tags_commandprefix: \!
+tags_commandatstart: True
```
Comment (GregSutcliffe) [inline base-config.yaml:22]: The "!" is confusing me, we're using "^" right now, so I think I'm mixing this up with the actual bot commands. This is *just* for the regex generation, right? Can we name it for that and maybe set a different default so it doesn't conflict?

**Review by GregSutcliffe — COMMENTED** [review]
Looks awesome @ryanlerch! Left a couple of comments but I'm going to merge this to my dev setup today and test it out.

File: meetings/__init__.py (hunk 1 above)
Comment (ryanlerch) [reply meetings/__init__.py]: sorry, yeah these went trhough a couple of iterations. will change them in both places to tag_commandprefix and tag_commandatstart

File: base-config.yaml (hunk 2 above)
Comment (ryanlerch) [reply base-config.yaml:22]: yeah, this is just for the tags -- i'll change it back to \^ here. i'll put a note that this is used in a regex and needs to be escaped if it has to be.

Comment (ryanlerch) [reply base-config.yaml:22]: in Fedora, i kind of want all the commands (including the tag ones) to be !action etc, that is why i made it a setting. :)

File: meetings/__init__.py (hunk 1 above)
Comment (ryanlerch) [reply meetings/__init__.py]: ok got a bit confused there. updated the helper.copy bit to the right keys:

    helper.copy("tags")
    helper.copy("tags_commandprefix")
    helper.copy("tags_commandatstart")

File: base-config.yaml (hunk 2 above)
Comment (GregSutcliffe) [reply base-config.yaml:22]: Yeah, I can see where you're going, but I think for defaults it helps to highlight the difference between the two. Having it configurable is sensible though. Thanks!

File: meetings/__init__.py (hunk 1 above)
Comment (GregSutcliffe) [reply meetings/__init__.py]: I still prefer the snake_case command arguments, they're way more readable. I know I'm bikeshedding here, but if we change them later we (in theory) should then write a config migration as well, which is a pain. Happy to make the change while merging if you have no objections? No need to push more code :)

Comment (ryanlerch) [reply meetings/__init__.py]: no objections!

**Comment by GregSutcliffe** [comment]
Merged as ed2454f28899f1fb210f18f33c8d7cb81603e644,  thanks so much @ryanlerch :tada:

**Pull request closed without merging**